pkg/operator/ceph/object/user/controller.go

```diff
@@ -225,25 +331,47 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 		StringData: secrets,
 		Type:       k8sutil.RookType,
 	}
-	k8sutil.SetOwnerRef(&secret.ObjectMeta, &c.ownerRef)
 
-	_, err = context.Clientset.CoreV1().Secrets(u.Namespace).Create(secret)
+	return secret
+}
+
+func (r *ReconcileObjectStoreUser) reconcileCephUserSecret(cephObjectStoreUser *cephv1.CephObjectStoreUser) (reconcile.Result, error) {
+	// Generate Kubernetes Secret
+	secret := r.generateCephUserSecret(cephObjectStoreUser)
+
+	// Set owner ref to the object store user object
+	err := controllerutil.SetControllerReference(cephObjectStoreUser, secret, r.scheme)
 	if err != nil {
-		return errors.Wrapf(err, "failed to save user %s secret", u.Name)
+		return reconcile.Result{}, errors.Wrapf(err, "failed to set owner reference for ceph object user %q secret", secret.Name)
 	}
-	logger.Infof("created user %s", u.Name)
-	return nil
+
+	// Create Kubernetes Secret
+	_, err = r.context.Clientset.CoreV1().Secrets(cephObjectStoreUser.Namespace).Create(secret)
+	if err != nil {
+		if kerrors.IsAlreadyExists(err) {
+			_, err := r.context.Clientset.CoreV1().Secrets(cephObjectStoreUser.Namespace).Update(secret)
```

Comment:
Will we ever be updating the user creds? Seems like we wouldn't need to update the secret after it's created.

Reply:
No but if someone touches the secret then the reconcile will reset it.

Reply:
I wonder how much it is worth to try and recover from scenarios like that where the admin is modifying the system. IMO it's not supported since there are too many configurations like that to worry about. It's fine, but just seems like the tip of the iceberg.